### PR TITLE
478 Call Stack Document

### PR DESCRIPTION
**Pull request by ACStoneCL**
### Related links

[Call Stack Example - Document the explanation #478](https://github.com/casper-network/docs/issues/478)

### Changes

Created a new document explaining the concept and functionality of call stacks within the `Building dApps` section of the documentation.

### Notes

Ran locally without issue.

### CODE REVIEW

**Review by deuszex — APPROVED** [review]

File: source/docs/casper/dapp-dev-guide/building-dapps/callstack.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+When the session code within a Deploy interacts with one or more contracts, this is the beginning of a [`Call Stack`](https://docs.rs/casper-types/1.5.0/casper_types/system/enum.CallStackElement.html). A call stack is the chronological order in which contracts call other contracts, initiated by an instance of session code.
+
+## The Caller
+
+In every instance of a call stack, the originating [caller](https://docs.rs/casper-types/1.5.0/casper_types/system/mint/trait.RuntimeProvider.html#tymethod.get_caller) is the session code within the account's context that began the interaction. Contract code cannot spontaneously act without session code to activate it. As such, the session code represents the *zeroth* entity in each call stack.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/building-dapps/callstack.md:9]: Using "latest" in the link will resolve to the latest version of the docs.
```suggestion
In every instance of a call stack, the originating [caller](https://docs.rs/casper-types/latest/casper_types/system/mint/trait.RuntimeProvider.html#tymethod.get_caller) is the session code within the account's context that began the interaction. Contract code cannot spontaneously act without session code to activate it. As such, the session code represents the *zeroth* entity in each call stack.
```

File: source/docs/casper/dapp-dev-guide/building-dapps/callstack.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Understanding Call Stacks
+
+Users wishing to interact with a Casper network must do so through [sending a Deploy](/dapp-dev-guide/building-dapps/sending-deploys/). All Deploys consist of [session code](/dapp-dev-guide/writing-contracts/session-code/) run in the context of the user account that sent the Deploy. The session code may [install contract code to global state](/dapp-dev-guide/writing-contracts/installing-contracts/), or interact with previously [installed contract code](/dapp-dev-guide/building-dapps/calling-contracts/).
+
+When the session code within a Deploy interacts with one or more contracts, this is the beginning of a [`Call Stack`](https://docs.rs/casper-types/1.5.0/casper_types/system/enum.CallStackElement.html). A call stack is the chronological order in which contracts call other contracts, initiated by an instance of session code.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/building-dapps/callstack.md:5]:
```suggestion
When the session code within a Deploy interacts with one or more contracts, this is the beginning of a [`Call Stack`](https://docs.rs/casper-types/latest/casper_types/system/enum.CallStackElement.html). A call stack is the chronological order in which contracts call other contracts, initiated by an instance of session code.
```

File: source/docs/casper/dapp-dev-guide/building-dapps/callstack.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+In every instance of a call stack, the originating [caller](https://docs.rs/casper-types/1.5.0/casper_types/system/mint/trait.RuntimeProvider.html#tymethod.get_caller) is the session code within the account's context that began the interaction. Contract code cannot spontaneously act without session code to activate it. As such, the session code represents the *zeroth* entity in each call stack.
+
+## The Immediate Caller
+
+The [immediate caller](https://docs.rs/casper-types/1.5.0/casper_types/system/mint/trait.RuntimeProvider.html#tymethod.get_immediate_caller) is the caller that interacted directly with the contract in question.
+
+If session code calls a contract, which in turn calls another contract, then the session code would represent the *zeroth* entity in the stack, the contract called by the initiating session code would be the *first* and the contract called by the first contract would be the *second*.
+
+In this example, the first contract would be the `immediate caller` of the second contract. The session code would remain the `caller`.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/building-dapps/callstack.md:17]: A simple image would be very nice here. It's not a requirement or a blocker for this PR, but perhaps an enhancement if there is time.

**Review by ipopescu — APPROVED** [review]
Very nice!

**Comment by ipopescu** [comment]
@ACStoneCL, I like the diagram. I played with it a bit to resize it and center it. I think the PR is good to go. Thanks so much.

Screen Shot 2022-12-14 at 8 54 23 PM

Screen Shot 2022-12-14 at 8 53 43 PM

**Pull request merged**